### ISSUE TITLE
Improve the structure of this lib

### ISSUE THREAD

**Issue by hongquan** [fossasia/susi_python#4]
This is my idea about how it can be improved:

1. Get rid of `global`. This is not Pythonic. It may be useful for ad-hoc automation script, but for a serious, well-structured library, uhm, no.

2. Let accept server URL as input, so that the lib can be used with SUSI server installed anywhere. We have these URLs hardcoded: `https://api.susi.ai`, `http://localhost:4000/`, but we should treat them as "default" value. Actually, when we make server URL customizable, we only need 1 default value, either `https://api.susi.ai` or `http://localhost:4000/`.

3. If we can, let's make this lib "sans-IO", i.e, the lib won't do the job of receiving, sending data (over network), it only takes care of parsing, combining data. With this approach, our lib can be used both in synchronous code (where [`requests`](https://pypi.org/project/requests/) is often used for transporting) and asyncio code (where [`aiohttp`](https://pypi.org/project/aiohttp/) is used for transporting). List of sans-io libs as reference: https://github.com/brettcannon/sans-io/blob/master/index.rst

**Comment by norbusan** [comment]
I think 1 and 2 are relatively easy and can be implemented without big interruption.

3 sounds a bit more complicated, though, since it requires adaptions of `susi_linux` interaction, I guess.

I vote for getting 1 and 2 done soon!

**Comment by stealthanthrax** [comment]
@hongquan , we can use env_variables for 2 instead of taking input from the user. It will make it simpler to use in susi linux as well

**Comment by hongquan** [comment]
@stealthanthrax "taking input",  I meant passing it as function/class parameter, not that user has to type into terminal.

The lib should not rely only on env variable. Env variable should be only one of methods that the lib supports.

**Comment by norbusan** [comment]
Agreed concerning function parameters and "only one method", thanks!

**Comment by stealthanthrax** [comment]
@hongquan @norbusan , I'll be fixing part 1 and 2 today

**Comment by hongquan** [comment]
@stealthanthrax Great!
